update-config-files.php

```diff
@@ -0,0 +1,369 @@
+        );
+        $portalConfigs = rGlob('./*config.php');
+        foreach ($portalConfigs as $config)
+        {
+            if (preg_match('/\\/LEAF_Request_Portal\\/\\w.*/', $config))
+            {
+                //make a backup first
+                $date = new Datetime();
+                $timestamp = $date->format('U');
+                $backupConfig = dirname($config) . '/' . basename($config) . '.' . $timestamp . '.php';
+                copy($config, $backupConfig);
+
+                $file = file_get_contents($config);
+                $varArray = parseStr($file);
+                $vars = array_intersect_key(array_merge_recursive($portalConfig, $varArray), $portalConfig);
+                $savedConfig = savePortal($config, $vars);
+                //print_r($savedConfig);
+            }
+        }
+    }
+
+    /**
+     * [writeNexusConfig writes/backup all Nexus configs in a root dir]
+     * To add a variable to the config files add it to the array or to remove a variable remove it from the array
+     * Note: if no value is supplied for a variable it will use the one in config file if exists
+     * @return [string] [config content]
+     */
+    function writeNexusConfig()
+    {
+        $nexusConfig = array(
+          'title' => array('varName' => 'title', 'varValue' => '', 'varComment' => '', 'varType' => ''),
+          'city' => array('varName' => 'city', 'varValue' => '', 'varComment' => ''),
+          'adminLogonName' => array('varName' => 'adminLogonName', 'varValue' => '', 'varComment' => 'Administrator\'s logon name', 'varType' => ''),
+          'adPath' => array('varName' => 'adPath', 'varValue' => '', 'varComment' => 'Active directory paths', 'varType' => ''),
+          'uploadDir' => array('varName' => 'uploadDir', 'varValue' => '', 'varComment' => 'Directory for user uploads. using backslashes (/), with trailing slash', 'varType' => ''),
+          'ERM_Sites' => array('varName' => 'ERM_Sites', 'varValue' => "array('')", 'varComment' => 'URL to ERM sites with trailing slash', 'varType' => ''),
+          'leafSecure' => array('varName' => 'leafSecure', 'varValue' => 'false', 'varComment' => 'leafSecure', 'varType' => ''),
+          'dbHost' => array('varName' => 'dbHost', 'varValue' => '', 'varComment' => '', 'varType' => ''),
+          'dbName' => array('varName' => 'dbName', 'varValue' => '', 'varComment' => '', 'varType' => ''),
+          'dbUser' => array('varName' => 'dbUser', 'varValue' => '', 'varComment' => '', 'varType' => ''),
+          'dbPass' => array('varName' => 'dbPass', 'varValue' => '', 'varComment' => '', 'varType' => ''),
+          'myNewVar' => array('varName' => 'myNewVar', 'varValue' => "'myNewValue'", 'varComment' => 'my new comment', 'varType' => 'static'),
+        );
+
+        $nexusConfigs = rGlob('./config.php');
+        foreach ($nexusConfigs as $config)
+        {
+            if (preg_match('/\\/LEAF_Nexus\\/\\w.*/', $config))
```

Comment:
The folder won't necessarily be named LEAF_Nexus. Should be a way to find the folder name based off the location of db_config.php

Reply:
Done